### ISSUE TITLE
[TESTFAIL] rspamd_fstring_gzip failing with ENABLE_JEMALLOC=ON

### ISSUE THREAD

**Issue by ln5** [rspamd/rspamd#4593]
```
make run-test
...
TEST SUITE: rspamd_utils
TEST CASE:  rspamd_fstring_gzip tests
  Some real compression

/path/to/rspamd/test/rspamd_cxx_unit_utils.hxx:132: ERROR: CHECK( memcmp(fstr->str, "helohelo", sizeof("helohelo") - 1) == 0 ) is NOT correct!
  values: CHECK( 1 == 0 )
```

OS: Fedora 38
compiler 1: clang version 16.0.6 (Fedora 16.0.6-2.fc38)
compiler 2: gcc (GCC) 13.2.1 20230728 (Red Hat 13.2.1-1)
CPU arch: x86_64
rspamd version: git master (3.6-95-g3744f8326)
cmake arguments: -DENABLE_JEMALLOC=ON
jemalloc package info:
```
Version      : 5.3.0
Release      : 3.fc38
Source       : jemalloc-5.3.0-3.fc38.src.rpm
```

Test fails with both clang and gcc.

**Comment by vstakhov** [comment]
Yes, it seems to be broken but I'm really surprised that it ever works sometimes :)

**Comment by vstakhov** [comment]
I believe it is fixed by 0b51255f334ced4362257fc2c93a48489c2ec55e

**Comment by ln5** [comment]
Indeed -- master works fine now. Thanks.
